molecule/default/molecule.yml

```diff
@@ -10,22 +10,39 @@ platforms:
   - name: bionic
     image: ubuntu:bionic
   - name: xenial
-    image: ubuntu:xenial
+    image: solita/ubuntu-systemd:xenial
+    privileged: True
+    command: /sbin/init
   - name: trusty
     image: ubuntu:trusty
   - name: centos6
     image: centos:6
   - name: centos7
-    image: centos:7      
+    image: milcom/centos7-systemd
+    privileged: True
 provisioner:
   name: ansible
   env:
-    ANSIBLE_ROLES_PATH: $HOME/wazuh-ansible/roles
+    ANSIBLE_ROLES_PATH: ../../roles
   lint:
     name: ansible-lint
     enabled: true # fix in seperate PR
 scenario:
   name: default
+  test_sequence:
+    - lint
+    - dependency
+    - cleanup
```

Comment:
Hi @dj-wasabi , if I keep this `cleanup` value, I'm getting the following error:
```
AttributeError: 'module' object has no attribute 'cleanup'
```

Reply:
What version of `molecule` is installed on your host? I had a successful run on my Mac with version: `molecule==2.20.1`

Reply:
Same version here:
```
# molecule --version
molecule, version 2.20.1
```
Also, I've noticed that when I remove the `cleanup` feature from the `molecule.yml`, the process freezes when doing the `Gathering facts` of Centos 7. Additionally, if I just choose to deploy Ubuntu Xenial and Bionic, it returns this warning:  

```
 TASK [Install dependencies] ****************************************************
 [WARNING]: Error deleting remote temporary files (rc: -9, stderr: })
FAILED - RETRYING: Install dependencies (3 retries left).
    changed: [xenial]
```
And freezes at that point again.

Reply:
Hi. The `'cleanup'` is resolved. The `pipenv` uses a different version of `molecule` which didn't had the `cleanup` task. I pushed a new `Pipfile.lock` file that contains newer versions for example Molecule.

The freezing is something I don't understand yet. I've tested it also on my work Mac, but no issues there. One thing though, I have updated the Docker properties on both my Mac's with increased memory size to at least 8 - 10GB (That was already long ago configured before I started with this, so wouldn't say that it could be a solution..) Can you restart your Docker Service and see if that helps?

Reply:
@manuasir Where you able to resolve your issue?

Reply:
Sorry for the late response, @dj-wasabi . I wasn't able to resolve that issue, I will try that again in another operating system and will let you know.

Reply:
Finally, I was able to execute the tasks in Ubuntu. But I've noticed that I get plenty of `skipped` and `changed` actions when executing the tasks. Do you know the reason for it? For example:
```
    TASK [wazuh/ansible-wazuh-manager : RedHat/CentOS/Fedora | Install Wazuh repo] ***
    skipping: [trusty]
    skipping: [bionic]
    skipping: [xenial]
    ok: [centos7]
    ok: [centos6]
```

or:

```
    TASK [wazuh/ansible-wazuh-manager : RedHat/CentOS/Fedora | Install openscap] ***
    skipping: [trusty] => (item=openscap-scanner) 
    skipping: [trusty] => (item=openssl) 
    skipping: [bionic] => (item=openscap-scanner) 
    skipping: [bionic] => (item=openssl) 
    skipping: [xenial] => (item=openscap-scanner) 
    skipping: [xenial] => (item=openssl) 
    changed: [centos7] => (item=openscap-scanner)
    changed: [centos6] => (item=openscap-scanner)
    ok: [centos6] => (item=openssl)
    changed: [centos7] => (item=openssl)
```

Also, I've seen that some tasks fail _sometimes_ :
```
    TASK [wazuh/ansible-wazuh-manager : Install wazuh-manager, wazuh-api and expect] ***
    FAILED - RETRYING: Install wazuh-manager, wazuh-api and expect (3 retries left).
```

I commented several platforms in the `molecule.yml` file and left _CentOS 7_.  I've been able to see the Testinfra battery test running only in that case. Otherwise, some tasks fail. The new tests look amazing, by the way :) 

Thanks

Reply:
Hi @manuasir 

Nice, that you where able to get it working. In the end, you don't have to do anything in the `molecule.yml` to disable platforms, all are executed. During development on specific tasks for RedHat platform, I disable the others so it is easier and faster to develop.

But your questions:

```
TASK [wazuh/ansible-wazuh-manager : RedHat/CentOS/Fedora | Install Wazuh repo] ***
skipping: [trusty]
skipping: [bionic]
skipping: [xenial]
ok: [centos7]
ok: [centos6]
```
Some tasks are `skipping`, some `ok`, some `changed`. In the above case, you'll see some `skipping` and that is fine. Based on what can be found in the `when` configuration off the task, it will be skipped on non-RedHat systems. So the task is doing something specific for RedHat (Installing packages with yum) which won't work on Ubuntu and Debian.

So `skipping` is always related to the `when` configuration of a task. `Changed` is when the task has something changed on the host. This could be installation of a package, placing of a file, chmoding of a file etc. The `ok` means that no change was done.

This is also the case for your 2nd example. Runs a task that only can be done on RedHat systems and not on Ubuntu.

The 3rd and last example is not really failing.
```
TASK [wazuh/ansible-wazuh-manager : Install wazuh-manager, wazuh-api and expect] ***
FAILED - RETRYING: Install wazuh-manager, wazuh-api and expect (3 retries left).
```
As part of this PR and based on what Ansible suggests (They also check this with the `ansible-lint` tool when you import a role in the Ansible Galaxy) is that you should retry tasks that work with "network" related services. There can always be a hickup on the host, network or ... and normally the task fails. But there is a retry mechanism for the task and it will execute the task 3 times until the playbook fails. In the above example, the task failed once and then retried it and then it worked.

So `FAILED - RETRYING` means that the first try failed and Ansible and a 2nd try is executed. Because I don't see any other log in your example I assume the 2nd try was successful.

Reply:
Thanks, @dj-wasabi